storageClient/storageClient.go

```diff
@@ -213,6 +213,60 @@ func (c *StorageClient) EncryptMembershipKeyForGroupMember(ctx context.Context,
 
 }
 
+// GetGroupMembershipKey returns member information about the group
+func (c *StorageClient) GetGroupMembershipKey(ctx context.Context, params GetMembershipKeyRequest) (*ClientGroup, error) {
+	var result *ClientGroup
+	path := c.Host + storageServiceBasePath + "/groups/" + params.GroupID.String() + "/membership_key"
+	req, err := e3dbClients.CreateRequest("GET", path, params)
+	if err != nil {
+		return result, err
+	}
+	err = e3dbClients.MakeSignedServiceCall(ctx, c.requester, req, c.SigningKeys, c.ClientID, &result)
+	return result, err
+}
+
+// CreateGroupAccessKey
+func (c *StorageClient) CreateGroupAccessKey(ctx context.Context, params GroupAccessKeyRequest) (e3dbClients.SymmetricKey, string, error) {
+	var wrappedAccessKey string
+	var accessKey e3dbClients.SymmetricKey
+	// Decrypt Access Key
+	rawEncryptionKey, err := e3dbClients.DecodeSymmetricKey(c.EncryptionKeys.Private.Material)
+	if err != nil {
+		return accessKey, wrappedAccessKey, err
+	}
+	accessKey, err = e3dbClients.DecryptEAK(params.EncryptedAccessKey, c.EncryptionKeys.Public.Material, rawEncryptionKey)
+	if err != nil {
+		return accessKey, wrappedAccessKey, err
+	}
+	// Encrypt Access Key with Group's Public Key
+	groupPubKey := params.PublicKey
+	encryptionKeys := e3dbClients.EncryptionKeys{
+		Public: e3dbClients.Key{
+			Type:     e3dbClients.DefaultEncryptionKeyType,
+			Material: groupPubKey,
+		},
+		Private: c.EncryptionKeys.Private,
```

Comment:
I think the key to wrap should be the access key for this record, not the clients private key. Keep in mind how powerful and sensitive a clients private key is, if that gets compromised all their data could be read, versus if only an access key for a record is leaked only records of that type are compromised.